[feat] use the IPFS proxy

melMass opened the pull request.

xat committed:
[feat] ⚡ add IPFS proxy support (#136)

Co-authored-by: Mel Massadian <melmassadian@gmail.com>

melMass committed:
fix: ⚡️ use rocks end point

melMass committed:
feat: ⚡️ wip: replaced calls in minter

melMass committed:
fix: 🎨 fix build warnings

melMass committed:
Merge branch 'main' into mel/feat/use-ipfs-proxy

melMass committed:
fix: ⚡️ thumbnail + console.log refactoring

melMass committed:
fix: ⚡️ clearer name for custom license URI field

melMass commented:
~~As discussed on Discord, CID versions might be an issue 😫~~, 

NFT.storage's CID has a toV0() method.

![image](https://user-images.githubusercontent.com/7041726/182596051-b0c8748e-ee08-4e13-8e88-92fc481ca9a1.png)

melMass committed:
fix: ⚡️ don't store rights uri if license is not custom

Zir0h commented on src/data/ipfs.js:
```diff
@@ -308,5 +307,10 @@ async function uploadMetadataFile({
   }
 
+  if (rights === 'custom') {
+    metadata.rightsUri = rightUri
```
rightsUri or rightUri? I get confused :sweat_smile: 
// edit: sorry, didn't see the "not ready for review yet" msg

melMass replied:
> rightsUri or rightUri? I get confused 😅

Hehe good point, we should match them all, on the indexer it's using what for now?

Zir0h replied:
The indexer uses `rightUri`

melMass committed:
fix: ⚡️ use rightUri everywhere

Instead of rightsUri

melMass committed:
fix: ⚡️ mint

melMass committed:
fix: ⚡️ arguments type (blob)

melMass committed:
fix: ⚡️ small fixes

melMass committed:
fix: ⚡️ use File

melMass committed:
fix: ⚡️ remove unused

melMass committed:
Merge branch 'main' into mel/feat/use-ipfs-proxy

melMass committed:
fix: 🎨  removes language from meta if not present

like righturi

melMass committed:
fix: ⚡️ fullscreen preview

melMass committed:
fix: ⚡️ typo

melMass committed:
chore: 🚸 json doc | code splitting

melMass committed:
test: 🚧 allow pointer lock

melMass committed:
fix: ⚡️ typo cid-> uri

melMass committed:
chore: ➖ cleanup

melMass committed:
fix: ⚡️ validation for restored fields

melMass committed:
fix: ⚡️ markdown preview

melMass committed:
fix: ⚡️ remove cover on successful mint

As now mint is properly handled thoughout, the mint component
is not unmounted on successive mints. This clears the cover
for these cases.

melMass committed:
Merge branch 'main' into mel/feat/use-ipfs-proxy

melMass committed:
fix: ⚡️ clear artifact and cover on success

melMass committed:
fix: ⚡️ tags can be null

melMass committed:
fix: ⚡️ use mime types

melMass committed:
fix: ⚡️ prepend names like in formats

melMass commented:
Note for later:

- Paths for subfolders are "screwed" in the utils/html zip / unzip phases.
- Mimetypes is defined too many times (not a big deal)

melMass committed:
fix: ⚡️ for empty tags + debug
